hash_practice/exercises.py

```diff
@@ -2,28 +2,75 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n^2)
+        Space Complexity: O(n)
     """
-    pass
+    if not strings:
+        return []
+    anagrams = {}
+    for string in strings:
+        sorted_string = "".join(sorted(string))
+        if sorted_string in anagrams:
+            anagrams[sorted_string].append(string)
+        else:
+            anagrams[sorted_string] = [string]
+    return list(anagrams.values())
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)

```

Comment:
Since this solution uses `sorted` on line 33, our overall time complexity would be `O(n)` to fill `counts` + `sorted`'s complexity `O(n*log(n))`. That gives us `O(n + n*log(n))`, which we'd simplify to `O(n*log(n))`.